test/logformats_test.py

```diff
@@ -651,6 +651,33 @@ def test_write_millisecond_handling(self):
 
         self.assertEqual(expected_file.read_text(), actual_file.read_text())
 
+    def test_write(self):
+        now = datetime(
+            year=2017, month=9, day=30, hour=15, minute=6, second=13, microsecond=191456
+        )
+
+        # We temporarily set the locale to C to ensure test reproducibility
+        with override_locale(category=locale.LC_TIME, locale_str="C"):
+            # We mock datetime.now during ASCWriter __init__ for reproducibility
+            # Unfortunately, now() is a readonly attribute, so we mock datetime
+            with patch("can.io.asc.datetime") as mock_datetime:
+                mock_datetime.now.return_value = now
+                writer = can.ASCWriter(self.test_file_name)
+
+            msg = can.Message(
+                timestamp=now.timestamp(),
+                arbitration_id=0x123,
+                data=(_ for _ in range(64)),
```

Comment:
Does this work, too?

```suggestion
                data=range(64),
```